resources/shellHelperFunctions.sh

```diff
@@ -342,9 +342,16 @@ builder_do_typescript_tests() {
   if [[ $# -gt 0 ]]; then
     C8_THRESHOLD=$1
     THRESHOLD_PARAMS="--lines $C8_THRESHOLD --statements $C8_THRESHOLD --branches $C8_THRESHOLD --functions $C8_THRESHOLD"
+  else
+    # Seems like a bug in TeamCity reporter if we don't list default thresholds,
+    # see #13418.
+    #
+    # Making branch and function thresholds slightly lower, because the default
+    # for c8 is 0 for these anyway.
+    THRESHOLD_PARAMS="--lines 90 --statements 90 --branches 80 --functions 80"
   fi
 
-  c8 --reporter=lcov --reporter=text --exclude-after-remap --check-coverage=false $THRESHOLD_PARAMS mocha ${MOCHA_FLAGS} "${builder_extra_params[@]}"
+  c8 --reporter=lcov --reporter=text --exclude-after-remap --check-coverage $THRESHOLD_PARAMS mocha ${MOCHA_FLAGS} "${builder_extra_params[@]}"
```

Comment:
Are there any tests where we want `--check-coverage=false`?

Reply:
No, that was only there for one project (I forget which) which had low coverage. Now we just give it a low threshold instead, and as you can see with the tweaks I have made, all builds pass, so the coverage thresholds are good.